src/main/java/com/google/devtools/build/lib/analysis/platform/ConstraintCollection.java

```diff
@@ -42,11 +43,11 @@
   @Nullable private final ConstraintCollection parent;
   private final ImmutableMap<ConstraintSettingInfo, ConstraintValueInfo> constraints;
 
-  ConstraintCollection(ImmutableList<ConstraintValueInfo> constraints) {
+  public ConstraintCollection(ImmutableList<ConstraintValueInfo> constraints) {
     this(null, constraints);
   }
 
-  ConstraintCollection(
+  public ConstraintCollection(
```

Comment:
Now that this is public please document the behavior difference between the overloads and the meaning of the parameters, particularly the nullable "parent".